tooling/miniutil/src/mock_write.rs

```diff
@@ -0,0 +1,67 @@
+use std::{io::Write, str::from_utf8, cell::RefCell, rc::Rc};
+
+use gen_minirust::libspecr::hidden::GcCompat;
+
+/// A buffer to mock a GcWrite object. 
+/// It is used to catch output from MiniRust code for testing.
+#[derive(Default, Clone)]
+pub struct MockWrite {
+    buffer: Rc<RefCell<Vec<u8>>>,
+}
+
+impl MockWrite {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Get all output lines as Strings.
+    pub fn into_strings(self) -> Vec<String> {
+        let slice = self.buffer.borrow();
+
+        let string = from_utf8(&slice).unwrap().to_string();
+
+        string.split("\n").map( 
+            |str| str.to_string() 
+        ).collect()
+    }
+}
+
+impl Write for MockWrite {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        self.buffer.borrow_mut().extend_from_slice(buf);
+
+        Ok(buf.len())
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
+    }
+}
+
+// Nothing within has anything to do with specr-lang. This points to nothing.
+impl GcCompat for MockWrite {
+    fn points_to(&self, _buffer: &mut std::collections::HashSet<usize>) { }
+
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
```

Comment:
```suggestion
```
I updated libspecr so that `as_any` does not have to be implemented everywhere any more.